apps/challenges/views.py

```diff
@@ -340,3 +343,56 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_all_submissions_of_challenge(request, challenge_pk, challenge_phase_pk):
+    """
+    Returns all the submissions for a particular challenge
+    """
+    # check if challenge exists or not.
+    challenge_model = get_model_object(Challenge)
+    challenge = challenge_model(challenge_pk)
+
+    # check if challenge phase exists or not.
+    try:
+        challenge_phase = ChallengePhase.objects.get(pk=challenge_phase_pk, challenge=challenge)
+    except ChallengePhase.DoesNotExist:
+        response_data = {'error': 'Challenge Phase {} does not exist'.format(challenge_phase_pk)}
+        return Response(response_data, status=status.HTTP_404_NOT_FOUND)
+
+    # check if challenge_host_team exists or not.
+    if is_user_a_host_of_challenge(user=request.user, challenge_pk=challenge_pk):
```

Comment:
@taranjeet Since we are using different serializers(`ChallengeSubmissionManagementSerializer` and `SubmissionSerializer`) based on the type of the user we are getting, I don't think there is a way to take that part of code out of the `if condition`. What do you say?